File: hint/src/main/java/datacomprojects/com/hint/TipsList.java
package datacomprojects.com.hint;

import android.content.Context;

import java.util.ArrayList;

import datacomprojects.com.hint.callbacks.TipNeedToDismissTipInterface;
import datacomprojects.com.hint.callbacks.TipShowCallback;

public class TipsList {

    private Context context;

    public TipsList(Context context) {
        this.context = context;
    }

    public ArrayList<Tip> tipArrayList = new ArrayList<>();

    public void setHintShowedCallback(TipShowCallback hintShowedCallback) {
        this.hintShowedCallback = hintShowedCallback;
    }

    private TipShowCallback hintShowedCallback;

    public void showNext(TipNeedToDismissTipInterface tipNeedToDismissTipInterface) {
        for (Tip tip :
                tipArrayList) {
                if(tip.showIfNeed(context)) {
                    tip.customHint.setTipNeedToDismissTipInterface(tipNeedToDismissTipInterface);
                    if(hintShowedCallback!=null)
                        hintShowedCallback.onShow(tip.hintID);
                    break;
                }
        }
    }

    public boolean allWasShowed() {
        for(Tip tip: tipArrayList)
            if(!tip.wasShowed(context))
                return false;
        return true;
    }

    public void dismissAll() {
        for(Tip tip : tipArrayList)
            tip.hide(context);
    }

    public void addAll() {
        for(Tip tip : tipArrayList)
            tip.reset(context);
    }

    public void dismissCurrent() {
        for(Tip tip : tipArrayList)
            if(tip.isShow) {
                tip.hide(context);
                if(hintShowedCallback!=null)
                    hintShowedCallback.onDismiss(tip.hintID);
                break;
            }
    }

}

File: hint/src/main/java/datacomprojects/com/hint/TipsList.kt
package datacomprojects.com.hint

import android.content.Context

import java.util.ArrayList

class TipsList(private val context: Context) {

    private var tipArrayList = ArrayList<Tip>()

    fun addTipToArray(tip: Tip) {
        tipArrayList.add(tip)
    }

    fun showNext() {
        for (tip in tipArrayList) {
            if (!tip.wasShowed(context)) {
                tip.showIfNeed(context) {
                    tip.hide(context) {
                        showNext()
                    }
                }
                break
            }
        }
    }

    fun allWasShowed(): Boolean {
        for (tip in tipArrayList)
            if (!tip.wasShowed(context))
                return false
        return true
    }

    fun dismissAll() {
        for (tip in tipArrayList)
            tip.hide(context)
    }

    fun addAll() {
        for (tip in tipArrayList)
            tip.reset(context)
    }

    fun skip(onDismiss: ((String) -> Unit)? = null) {
        for (tip in tipArrayList)
            if (tip.isShow) {
                tip.hide(context) {
                    onDismiss?.let { it(tip.hintID) }
                }
                break
            }
    }

}
